odl/space/fspace.py

```diff
@@ -133,53 +175,137 @@ def out_dtype(self):
-            evaluation
+        Notes
+        -----
+        If you specify ``vectorized=False``, the function is decorated
+        with a vectorizer, which makes two elements created this way
+        from the same function being regarded as *not equal*.
         """
-        if not callable(fcall):
-            raise TypeError('`fcall` {!r} is not callable'.format(fcall))
+        if fcall is None:
+            return self.zero()
         elif fcall in self:
             return fcall
         else:
+            if not callable(fcall):
+                raise TypeError('`fcall` {!r} is not callable'.format(fcall))
             if not vectorized:
-                fcall = vectorize(fcall)
+                if self.field == RealNumbers():
+                    otypes = ['float64']
+                elif self.field == ComplexNumbers():
+                    otypes = ['complex128']
+                else:
+                    otypes = []
+
+                fcall = vectorize(otypes=otypes)(fcall)
 
             return self.element_type(self, fcall)
 
+    def zero(self):
+        """Function mapping everything to zero.
+
+        This function is the additive unit in the function space.
+
+        Since `FunctionSpace.lincomb` may be slow, we implement this function
+        directly.
+        """
+        def zero_vec(x, out=None):
+            """Zero function, vectorized."""
+            if is_valid_input_meshgrid(x, self.domain.ndim):
+                out_shape = out_shape_from_meshgrid(x)
+            elif is_valid_input_array(x, self.domain.ndim):
+                out_shape = out_shape_from_array(x)
+            else:
+                raise TypeError('invalid input type')
+
+            if out is None:
+                return np.zeros(out_shape, dtype=self.out_dtype)
+            else:
+                # Need to go through an array to fill with the correct
+                # zero value for all dtypes
+                fill_value = np.zeros(1, dtype=self.out_dtype)[0]
+                out.fill(fill_value)
+
+        return self.element_type(self, zero_vec)
+
+    def one(self):
+        """Function mapping everything to one.
+
+        This function is the multiplicative unit in the function space.
+        """
+        def one_vec(x, out=None):
+            """One function, vectorized."""
+            if is_valid_input_meshgrid(x, self.domain.ndim):
+                out_shape = out_shape_from_meshgrid(x)
+            elif is_valid_input_array(x, self.domain.ndim):
+                out_shape = out_shape_from_array(x)
+            else:
+                raise TypeError('invalid input type')
+
+            if out is None:
+                return np.ones(out_shape, dtype=self.out_dtype)
+            else:
+                # Need to go through an array to fill with the correct
+                # zero value for all dtypes
+                fill_value = np.ones(1, dtype=self.out_dtype)[0]
```

Comment:
Think of strings and stuff like that. Apart from that, the objects generated by `np.dtype()` cannot be called, so that's not an option.